src/main/java/com/nvidia/spark/rapids/jni/SparkResourceAdaptor.java

```diff
@@ -79,12 +96,12 @@ public SparkResourceAdaptor(RmmEventHandlerResourceAdaptor<RmmDeviceMemoryResour
   }
 
   @Override
-  public long getHandle() {
+  public synchronized long getHandle() {
```

Comment:
Is it possible someone can get the handle and then we can close it underneath them? I understand the PR says it's not trying to fix all possible races, but just wondering.

Reply:
yes. Technically the API is public, but it is used as if it is a private API.